offload/DeviceRTL/src/Kernel.cpp

```diff
@@ -34,6 +34,7 @@ inititializeRuntime(bool IsSPMD, KernelEnvironmentTy &KernelEnvironment,
   mapping::init(IsSPMD);
   state::init(IsSPMD, KernelEnvironment, KernelLaunchEnvironment);
   allocator::init(IsSPMD, KernelEnvironment);
+  __init_thread_dst();
```

Comment:
Also, I think you can put this init into one of the init above, such as `state::init`, to save an explicit synchronization.